expression/builtin_other.go

```diff
@@ -159,6 +159,41 @@ type baseInSig struct {
 	hasNull      bool
 }
 
+// ReplaceColumnOfExpr replaces column of expression by another LogicalProjection.
+// This function is the same as ReplaceColumnOfExpr in rule_eliminate_projection.go and will be assigned by init() there.
+// We are doing this because import that package directly will cause dependency cycle.
+var ReplaceColumnOfExpr func(expr Expression, projExpr []Expression, schema *Schema) Expression
+
+// ReplaceColumnOfExprForIn replaces column of nonConstArgs in IN expression by another LogicalProjection.
+func ReplaceColumnOfExprForIn(sf *ScalarFunction, projExpr []Expression, schema *Schema) {
+	switch inFunc := sf.Function.(type) {
```

Comment:
Can we use `sf.Function.(*baseInSig)`?

Reply:
Seems we can't.
Embedding `*baseInSig` doesn't make it able to be cast to `baseInSig`.